lib/src/conflicts.rs

```diff
@@ -375,6 +378,44 @@ impl Conflict<Option<FileId>> {
     }
 }
 
+impl<T> Conflict<Option<T>>
+where
+    T: Borrow<TreeValue>,
+{
+    /// Convert a `Conflict<Option<TreeValue>>` to a `Conflict<Tree>`, if all
+    /// non-`None` terms are `TreeValue::Tree`. `None` terms are converted to
+    /// empty trees.
+    pub fn to_tree_conflict(
+        &self,
+        store: &Arc<Store>,
+        dir: &RepoPath,
+    ) -> Result<Option<Conflict<Tree>>, BackendError> {
+        let tree_id_conflict = self.maybe_map(|term| match term {
+            None => Some(None),
```

Comment:
I hope we can eventually make this logic clearer; I'll have to think if I have any ideas.